tests/envs/test_registration.py

```diff
@@ -188,112 +193,108 @@ def test_spec_with_kwargs():
 
 
 def test_missing_lookup():
-    registry = registration.EnvRegistry()
-    registry.register(id="Test-v0", entry_point=None)
-    registry.register(id="Test-v15", entry_point=None)
-    registry.register(id="Test-v9", entry_point=None)
-    registry.register(id="Other-v100", entry_point=None)
+    register(id="Test1-v0", entry_point=None)
+    register(id="Test1-v15", entry_point=None)
+    register(id="Test1-v9", entry_point=None)
+    register(id="Other1-v100", entry_point=None)
     try:
```

Comment:
is this construct a poor man's version of `with pytest.raises(error.DeprecatedEnv)` ?

Reply:
I think so, I wanted to make as few changes to the tests as possible, but might as well make some updates to them as well

Reply:
I'll keep this out of the scope of this PR for now, this was taken from the original code, and this PR is meant to primarily improve a specific aspect of registration, and keep the functionality otherwise. This can be updated later